src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -900,7 +902,8 @@ public void createStream(ContextInternal context, HttpClientRequestImpl req, Pro
 
   @Override
   public boolean isValid() {
-    return expirationTimestamp == 0 || System.currentTimeMillis() <= expirationTimestamp;
+    long now = System.currentTimeMillis();
+    return expirationTimestamp == 0 || now <= expirationTimestamp && (!options.isKeepAliveTTLEnabled() || options.getKeepAliveTTL() * 1000 > now - initialTimestamp);
```

Comment:
can you surround with parenthesis the two expressions right and left of `&&` to make it more clear ?